From 138432920ae4c69faaadccdbb8ede35751425e69 Mon Sep 17 00:00:00 2001
From: ChiaYen Kan <kcygamil@gmail.com>
Date: Fri, 19 Jul 2024 08:50:24 +0800
Subject: [PATCH] fix link error in msys2

---
 configure.ac | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/configure.ac b/configure.ac
index 98439946..f3de14e9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -208,6 +208,15 @@ if test "x$have_cppunit" != "xyes"; then
   AC_MSG_WARN([$CPPUNIT_PKG_ERRORS])
 fi
 
+# Check iconv
+AC_CHECK_LIB([iconv], [iconv_open], , [ac_found_iconf=no])
+if test "x$ac_found_iconf" = "xno"; then
+  AC_CHECK_LIB([iconv], [libiconv_open], , [ac_found_iconf=no])
+fi
+if test "x$ac_found_iconf" = "xyes"; then
+  LIBS="-liconv $LIBS"
+fi
+
 # Check availability of libz
 if test "x$with_libz" = "xyes"; then
   PKG_CHECK_MODULES([ZLIB], [zlib >= 1.2.3], [have_zlib=yes], [have_zlib=no])
-- 
2.43.0.windows.1

