From ce3ce9aeb7add194844949b193117bc9f3132173 Mon Sep 17 00:00:00 2001
From: ChiaYen Kan <kcygamil@gmail.com>
Date: Tue, 23 Jul 2024 15:01:03 +0800
Subject: [PATCH] fix check file size error when build with msys2

---
 src/a2io.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/a2io.h b/src/a2io.h
index 27ddc518..e3876570 100644
--- a/src/a2io.h
+++ b/src/a2io.h
@@ -141,7 +141,7 @@
 #  define a2open(path, flags, mode) _wsopen(path, flags, _SH_DENYNO, mode)
 #  define a2fopen(path, mode) _wfsopen(path, mode, _SH_DENYNO)
 // # define a2ftruncate(fd, length): We don't use ftruncate in Mingw build
-#  define a2_off_t off_t
+#  define a2_off_t off64_t
 #elif defined(__ANDROID__) || defined(ANDROID)
 #  define a2lseek(fd, offset, origin) lseek64(fd, offset, origin)
 // # define a2fseek(fp, offset, origin): No fseek64 and not used in aria2
-- 
2.43.0.windows.1

